vowpalwabbit/core/include/vw/core/confidence_sequence.h

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#pragma once
+
+#if !defined(__APPLE__) && !defined(_WIN32)
+#  define __STDCPP_MATH_SPEC_FUNCS__ 201003L
+#  define __STDCPP_WANT_MATH_SPEC_FUNCS__ 1
+#endif
+
+#include "vw/core/io_buf.h"
+#include "vw/core/metric_sink.h"
+
+#include <algorithm>
+
+constexpr float DEFAULT_ALPHA = 0.05f;
+
+namespace VW
+{
+namespace confidence_sequence
+{
+struct IncrementalFsum
+{
+  std::vector<double> partials;
+
+  IncrementalFsum operator+=(double x)
+  {
+    int i = 0;
+    for (double y : this->partials)
+    {
+      if (std::abs(x) < std::abs(y)) { std::swap(x, y); }
+      double hi = x + y;
+      double lo = y - (hi - x);
+      if (lo != 0.0)
+      {
+        this->partials[i] = lo;
+        ++i;
+      }
+      x = hi;
+    }
+    this->partials[i] = x;
```

Comment:
isn't i out of range here?

Reply:
Good catch, added loop to ensure it is not out of range

Reply:
doesn't swapping this line with resizing work here?

Reply:
Yes that should work, updated